### PR TITLE
refactor: split ChainId useEffects

### PR DESCRIPTION
**Pull request by vm**
- Combines 3 effects into one (for syncing the chainId query param)
- Moves syncing query param logic into App.tsx
- Creates `useSelectChain` useCallback

### CODE REVIEW

File: src/components/Header/NetworkSelector.tsx [hunk 1]
```diff
@@ -425,7 +328,14 @@ export default function NetworkSelector() {
             </FlyoutHeader>
             {NETWORK_SELECTOR_CHAINS.map((chainId: SupportedChainId) =>
               isChainAllowed(connector, chainId) ? (
-                <Row onSelectChain={onSelectChain} targetChain={chainId} key={chainId} />
+                <Row
+                  onSelectChain={(targetChainId: SupportedChainId) => {
+                    selectChain(targetChainId)
+                    closeModal()
```
Comment (zzmp) [inline src/components/Header/NetworkSelector.tsx:334]: Should this wait until the selectChain callback resolves?

File: src/pages/App.tsx [hunk 2]
```diff
@@ -114,6 +115,8 @@ export default function App() {
   useAnalyticsReporter()
   initializeAnalytics()
 
+  useSyncChainQuery()
```
Comment (zzmp) [inline src/pages/App.tsx:118]: Can this still be in the network selector so that it's not global, but grouped with like logic?

**Review by zzmp — APPROVED** [review]
lgtm, just some minor questions

File: src/components/Header/NetworkSelector.tsx (hunk 1 above)
Comment (vm) [reply src/components/Header/NetworkSelector.tsx:334]: yeah i can change that

File: src/pages/App.tsx (hunk 2 above)
Comment (vm) [reply src/pages/App.tsx:118]: yeah i can move it back

**Pull request merged**